Shape_detection/include/CGAL/Shape_detection/Region_growing/Region_growing.h

```diff
@@ -376,26 +376,89 @@ namespace internal {
           // Check global conditions.
           if (!is_success || !m_region_type.is_valid_region(region)) {
             revert(region);
-          } else {
-            *(region_out++) = std::pair<typename RegionType::Primitive, Region>(m_region_type.primitive(), region);
+          }
+          else {
             fill_region_map(m_nb_regions++, region);
+            *region_out = std::make_pair(m_region_type.primitive(), std::move(region));
+            region_out++;
           }
         }
       }
 
       return region_out;
     }
 
+    /*!
+      \brief runs the region growing algorithm and fills an output iterator
+      with the fitted primitive and their region.
+
+      \tparam PrimitiveAndRegionOutputIterator
+      a model of `OutputIterator` whose value type is `Primitive_and_region`
+
+      \param region_out
+      an output iterator of type `PrimitiveAndRegionOutputIterator`.
+
+      \return past-the-end position in the output sequence
+    */
```

Comment:
Looks like the documentation here doesn't belong to this method, resulting in the overnight build (see: https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-5.6-Ic-196/logs_master/Shape_detection.log) also in
```
/home/cgal-testsuite/cgal_doc_build/CGAL-5.6-Ic-196/include/CGAL/Shape_detection/Region_growing/Region_growing.h:395: warning: CGAL::Shape_detection::Region_growing::detect has @param documentation sections but no arguments
/home/cgal-testsuite/cgal_doc_build/CGAL-5.6-Ic-196/include/CGAL/Shape_detection/Region_growing/Region_growing.h:395: warning: found documented return type for CGAL::Shape_detection::Region_growing::detect that does not return anything
```

Reply:
Is the function name (`detect`) covering what the function does, especially also seen the brief description "runs the region growing algorithm" or is the brief description incorrect?